File: SolaceApp/Views/Shared/Theme.swift
import SwiftUI
import SolaceCore

/// Shared visual language for the redesigned app: a small, consistent set of
/// colors, spacing, and corner-radius values so Liquid Glass surfaces read as
/// one system rather than being styled ad hoc per screen. Liquid Glass reads
/// best over colorful, dynamic content — flat white behind it looks dull —
/// so every top-level screen sits on an `AmbientBackground` rather than a
/// plain system background.
enum Theme {
    /// Primary brand accent — carried over from the original leaf mark.
    static let primary = Color.teal

    /// Secondary calming tone used on the Today and Games surfaces to
    /// differentiate "reflective" screens from the primary teal used for
    /// core actions (messaging, resources).
    static let secondary = Color(red: 0.62, green: 0.58, blue: 0.94) // soft lavender

    /// Warm tone reserved for streaks/encouragement, never for errors.
    static let warm = Color(red: 0.98, green: 0.71, blue: 0.42)

    /// A small rotating accent palette for iconography — quick-action tiles,
    /// game rows — so the app reads as lively rather than monochrome.
    static let coral = Color(red: 0.98, green: 0.45, blue: 0.42)
    static let sky = Color(red: 0.35, green: 0.68, blue: 0.98)
    static let leaf = Color(red: 0.40, green: 0.78, blue: 0.55)
    static let sun = Color(red: 0.98, green: 0.76, blue: 0.24)

    enum Spacing {
        static let xSmall: CGFloat = 4
        static let small: CGFloat = 8
        static let medium: CGFloat = 16
        static let large: CGFloat = 24
        static let xLarge: CGFloat = 32
    }

    enum Radius {
        /// Standard card corner radius, tuned to feel concentric with
        /// device corners at typical card insets.
        static let card: CGFloat = 24
        static let control: CGFloat = 16
    }

    /// Per-tab ambient gradient palettes (see `AmbientBackground`) — each
    /// tab gets its own gentle identity rather than one flat background.
    enum Ambient {
        static let today: [Color] = [sun, coral, warm, secondary, primary, sky, warm, coral, sun]
        static let talk: [Color] = [sky, primary, secondary, primary, sky, secondary, secondary, sky, primary]
        static let wellness: [Color] = [leaf, primary, sky, primary, leaf, secondary, sky, leaf, primary]
        static let games: [Color] = [secondary, coral, sun, primary, secondary, leaf, sun, secondary, coral]
        static let profile: [Color] = [primary, secondary, sky, secondary, primary, leaf, sky, primary, secondary]
    }
}

public extension MoodLevel {
    /// UI-layer color mapping — kept out of `SolaceCore` since that package
    /// has no SwiftUI dependency.
    var color: Color {
        switch self {
        case .awful: return Theme.coral
        case .bad: return Theme.warm
        case .okay: return Theme.sun
        case .good: return Theme.leaf
        case .great: return Theme.primary
        }
    }
}

File: SolaceAndroid/app/src/main/java/com/jonathanalumasa/solace/ui/theme/Theme.kt
package com.jonathanalumasa.solace.ui.theme

import android.os.Build
import androidx.compose.foundation.isSystemInDarkTheme
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.darkColorScheme
import androidx.compose.material3.dynamicDarkColorScheme
import androidx.compose.material3.dynamicLightColorScheme
import androidx.compose.material3.lightColorScheme
import androidx.compose.runtime.Composable
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.platform.LocalContext
import androidx.compose.ui.unit.dp

/** Spacing scale mirroring iOS `Theme.Spacing`. */
object Spacing {
    val xSmall = 4.dp
    val small = 8.dp
    val medium = 16.dp
    val large = 24.dp
    val xLarge = 32.dp
}

/** Corner radii mirroring iOS `Theme.Radius`. */
object Radius {
    val card = 24.dp
    val control = 16.dp
}

private val LightColors = lightColorScheme(
    primary = SolaceColors.Primary,
    secondary = SolaceColors.Secondary,
    tertiary = SolaceColors.Sky,
    error = SolaceColors.Coral
)

private val DarkColors = darkColorScheme(
    primary = SolaceColors.Primary,
    secondary = SolaceColors.Secondary,
    tertiary = SolaceColors.Sky,
    error = SolaceColors.Coral
)

/**
 * @param dynamicColor opts into Material You wallpaper-derived color on
 *   Android 12+. Defaults to off so the Solace palette stays recognisable
 *   across devices, matching the iOS app's fixed identity.
 */
@Composable
fun SolaceTheme(
    darkTheme: Boolean = isSystemInDarkTheme(),
    dynamicColor: Boolean = false,
    content: @Composable () -> Unit
) {
    val colorScheme = when {
        dynamicColor && Build.VERSION.SDK_INT >= Build.VERSION_CODES.S -> {
            val context = LocalContext.current
            if (darkTheme) dynamicDarkColorScheme(context) else dynamicLightColorScheme(context)
        }

        darkTheme -> DarkColors
        else -> LightColors
    }

    MaterialTheme(
        colorScheme = colorScheme,
        content = content
    )
}

/**
 * Per-tab ambient gradient palettes, mirroring iOS `Theme.Ambient` — each tab
 * gets its own gentle identity rather than one flat background. iOS renders
 * these as an animated mesh gradient; see `AmbientBackground` for how Compose
 * approximates that with drifting radial gradients.
 */
object Ambient {
    val today: List<Color> = listOf(
        SolaceColors.Sun, SolaceColors.Coral, SolaceColors.Warm, SolaceColors.Secondary
    )
    val talk: List<Color> = listOf(
        SolaceColors.Sky, SolaceColors.Primary, SolaceColors.Secondary, SolaceColors.Sky
    )
    val wellness: List<Color> = listOf(
        SolaceColors.Leaf, SolaceColors.Primary, SolaceColors.Sky, SolaceColors.Leaf
    )
    val games: List<Color> = listOf(
        SolaceColors.Secondary, SolaceColors.Coral, SolaceColors.Sun, SolaceColors.Leaf
    )
    val profile: List<Color> = listOf(
        SolaceColors.Primary, SolaceColors.Secondary, SolaceColors.Sky, SolaceColors.Leaf
    )
}
